front/lib/swr/user.ts

```diff
@@ -30,3 +33,44 @@ export function useUserMetadata(key: string) {
     mutateMetadata: mutate,
   };
 }
+
+export function useSearchMembers(
+  workspaceId: string,
+  searchTerm: string,
+  pageIndex: number,
+  pageSize: number
+) {
+  const searchMembersFetcher: Fetcher<SearchMembersResponseBody> = fetcher;
+  const debounceHandle = useRef<NodeJS.Timeout | undefined>(undefined);
+  const [debouncedSearchTerm, setDebouncedSearchTerm] = useState(searchTerm);
+
+  useEffect(() => {
+    const debouncedSearch = () => {
+      setDebouncedSearchTerm(searchTerm);
+    };
+
+    debounce(debounceHandle, debouncedSearch, 300);
+  }, [searchTerm]);
+
+  const searchParams = new URLSearchParams({
+    searchTerm: debouncedSearchTerm,
+    orderBy: "name",
+    lastValue: (pageIndex * pageSize).toString(),
+  });
+
+  const { data, error, mutate } = useSWRWithDefaults(
+    `/api/w/${workspaceId}/members/search?${searchParams.toString()}`,
+    searchMembersFetcher,
+    {
+      revalidateOnFocus: false,
+      revalidateOnReconnect: false,
+    }
+  );
+  return {
+    members: data?.members ?? [],
+    membersCount: data?.total ?? 0,
```

Comment:
We already have `totalNodesCount` can we stick to the same naming so `totalMembersCount`